### PR TITLE
build: add support for optionally building PythonKit

### PR DESCRIPTION
**Pull request by compnerd**
This adds support for building PythonKit as a package as part of the
toolchain build.

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves SR-NNNN.

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @shahmishal @Rostepher @edymtt @rxwei @dan-zheng

**Comment by compnerd** [comment]
@swift-ci please clean test

**Review by shahmishal — APPROVED** [review]
LGTM!

**Comment by shahmishal** [comment]
Can you add few tests for the new flag?

**Comment by Rostepher** [comment]
You can run the Python unit tests and lint via:

```
python utils/python_lint.py
python utils/build_swift/run_tests.py  # There are other flags as well, try --help
python utils/swift_build_support/run_tests.py
python utils/update_checkout/run_tests.py
```

File: utils/build_swift/build_swift/driver_arguments.py [hunk 1]
```diff
@@ -620,6 +621,9 @@ def create_argument_parser():
     option('--playgroundsupport', store_true('build_playgroundsupport'),
            help='build PlaygroundSupport')
 
+    option('--pythonkit', store_true('build_pythonkit'),
```
Comment (Rostepher) [inline utils/build_swift/build_swift/driver_arguments.py]: You'll likely need to add this to the test suite in `utils/build_swift/tests/expected_options.py`.

**Comment by compnerd** [comment]
@swift-ci please clean test

**Review by Rostepher — APPROVED** [review]

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please smoke test Linux platform

**Comment by compnerd** [comment]
@swift-ci please smoke test Linux platform

**Pull request merged**